### ISSUE TITLE
Yank all `0.3` versions older than `0.3.16`

### ISSUE THREAD

**Issue by faern** [rust-lang/socket2-rs#139]
Due to #120. If the standard library ever do change the memory layout of `SocketAddr`, usage of this crate of versions before `0.3.16` will have undefined behaviour and lead to segfaults. And even if the standard library never do change the layout of those types, those older versions of this crate still makes assumptions they should not.

This is exactly what yanking is for. Marking a release as "should not be used". I think all versions of `socket2` from `0.3.0` through `0.3.15` should be yanked from crates.io.

Related issues on other crates having the same problem: https://github.com/deprecrated/net2-rs/issues/107, https://github.com/tokio-rs/mio/issues/1412

**Comment by Thomasdezeeuw** [comment]
Done: https://crates.io/crates/socket2/versions. I'm leaving v0.1.0 and v0.2.0 although they might have the same problem (I haven't checked), but those aren't maintained any more.

**Comment by BWStearns** [comment]
Sorry for the necromancy, but if I have a project that uses this 0.3.11 through several dependencies that are all dead, do you have any suggestions for just getting it to compile? I've tried switching the rust versions back to 1.4x.0 (around the time of this original issue) but it doesn't seem to help. Getting it to compile at least would let me migrate to newer versions of the dependencies a bit easier.

```
    |
156 |         mem::transmute::<SocketAddrV4, sockaddr_in>(v4);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: `SocketAddrV4` (48 bits)
    = note: target type: `sockaddr_in` (128 bits)
```

**Comment by faern** [comment]
What happens if you try `cargo update -p socket2`? That should update `socket2` to `0.3.19` unless whatever crate depends on it has pinned a specific version.

What crate has the code you quote the error from? That code is invalid since those two types (no longer) have the same memory layout.

If you use `socket2 0.3.11` I'm pretty sure your dependency tree should be able to handle version `0.3.16` and up where this issue is fixed.

**Comment by Thomasdezeeuw** [comment]
@faern is entirely correct. v0.3.16 fixes this issue and is compatible with v0.3.x, while v0.4.x is not, so we don't want to yank that version.

**Comment by BWStearns** [comment]
For anyone in the future who runs across this thread (also possible future me): to fix this second issue the trick might be to set `export MACOSX_DEPLOYMENT_TARGET=11`.

@faern thank you so much! I had tried manually editing the versions up earlier but I must have missed something.
